src/app.js

```diff
@@ -118,6 +124,23 @@ function getOptions() {
   return options;
 }
 
+/**
+ * Because the rules engine talks to the server over the public HTTP/WS API,
+ * the gateway needs to configure it with a JWT and a server address
+ * @param {http.Server|https.Server} server
+ */
+function rulesEngineConfigure(server) {
```

Comment:
I left a comment above with a bit of the context of this. Where would you put the localBaseUrl? I think having a server to baseURL function would eliminate most of the complexity of this. `JSONWebToken.requestToken(-1)` is the `requestJWT` method.

There's definitely the potential for a bit of duplication. I think that having a wrapper of `node-fetch` that automatically acquires a JWT and is configured with the base URL would help considerably. It would eliminate the need for the intent parser to do anything fancy and would eliminate some amount of the Rules Engine's fetch code.